CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -547,20 +547,17 @@ end
 
 -- Makes the staff member request a raise of 10%, or a wage exactly in the middle of their current and a fair one, whichever is more.
 function Staff:requestRaise()
-  -- Is this the first time a member of staff is requesting a raise?
-  -- Only show the help if the player is playing the campaign though
-  if self.hospital and not self.hospital.has_seen_pay_rise and tonumber(self.world.map.level_number) then
-    self.world.ui.adviser:say(_A.information.pay_rise)
-    self.hospital.has_seen_pay_rise = true
-  end
+  assert(self.hospital, "A staff member asked for a pay rise who doesn't belong to a hospital!")
   -- Check whether there is already a request for raise.
   if not self:isMoodActive("pay_rise") then
     local amount = math.floor(math.max(self.profile.wage * 1.1, (self.profile:getFairWage() + self.profile.wage) / 2) - self.profile.wage)
     -- At least for now, staff are timid, and only ask for raises 1/5th of the time
     if math.random(1, 5) ~= 1 or amount <= 0 then
       self.timer_until_raise = nil
-      return
+      return -- too timid
     end
+    -- The staff member can now successfully ask for a raise
+    self.hospital:adviseRaiseRequest()
     self.quitting_in = 25*30 -- Time until the staff members quits anyway
     self:setMood("pay_rise", "activate")
     self.world.ui.bottom_panel:queueMessage("strike", amount, self)
```

Comment:
This can move too, outside of the if.

Reply:
If we move ``self.world.ui.bottom_panel:queueMessage("strike", amount, self)`` outside the ``if``` it's going to always queue a message even if the raise request random failed. And it won't know about "amount"

Reply:
Sorry, it can move to adviseRaiseRequest, eg line 420. Send amount as a param to adviseRaiseRequest, maybe self too (which would be called staff in PlayerHospital?)

Reply:
Maybe, I'm not great with function names.

Reply:
I suppose the other question is if PlayerHospital should be performing the queueMessage, and not staff? @Alberth289346